plugins/modules/packaging/os/pacman.py

```diff
@@ -411,9 +453,11 @@ def _build_install_diff(pacman_verb, pkglist):
             "after": "\n".join(sorted(after)) + "\n" if after else "",
         }
 
+        changed_reason_pkgs = [p for p in pkgs_to_set_reason if p not in installed_pkgs]
+
         if self.m.check_mode:
-            self.add_exit_infos("Would have installed %d packages" % len(installed_pkgs))
-            self.exit_params["packages"] = sorted(installed_pkgs)
+            self.add_exit_infos("Would have installed %d packages" % (len(installed_pkgs) + len(changed_reason_pkgs)))
+            self.exit_params["packages"] = sorted(installed_pkgs + changed_reason_pkgs)
```

Comment:
This is the only case, where the returned list of packages is sorted.
In my opinion, the returned list should be identical to the one returned if check mode is disabled.
The sorting was introduced in this commit: https://github.com/ansible-collections/community.general/pull/4286/commits/c28abd8200090641ea0fa4bdf569b7ad5706c624

Reply:
@jraby Is there any reason why you only sorted the package list in check mode and not in normal mode?

Reply:
I think it was an oversight on my part.
The idea of the sort was to make it easier to find a package, but maybe having the output in the same order as specified in the tasks would make more sense? (their dependencies would be sprinkled throughout  however)

But I don't care either way as long as check vs normal returns the same order. (the current situation makes no sense)

Reply:
As restoring the original order would require some effort and deciding over a strategy for dependencies and just combining the lists would result in an order hard to understand for outsiders, I decided to go for sorting in dfbd46d, especially as the diff is also sorted.

In the end, this may be irrelevant, since most users probably won't even look at the list and the other users might be experienced enough to use ansible's `sort`, `intersect` or `difference` feature to automatically get the desired results.